web/client/components/map/leaflet/Layer.jsx

```diff
@@ -45,6 +46,26 @@ const LeafletLayer = React.createClass({
         }
         this.updateLayer(newProps, this.props);
     },
+    shouldComponentUpdate(newProps) {
+        // the reduce returns true when a prop is changed
+        // optimizing when options are equal ignorning loading key
+        return !(["map", "type", "srs", "position", "zoomOffset", "onInvalid", "onClick", "options"].reduce( (prev, p) => {
```

Comment:
Honestly, I did not tested if it is perceptible in the normal application, but when you have both map and print preview (doubling in fact the number of features) it can become slow.